src/models/network.ts

```diff
@@ -0,0 +1,654 @@
+		const oldNetworkName = this.name;
+		const oldNick = this.nick;
+		const oldRealname = this.realname;
+
+		this.keepNick = null;
+		this.nick = args.nick;
+		this.host = String(args.host || "");
+		this.name = String(args.name || "") || this.host;
+		this.port = parseInt(args.port, 10);
+		this.tls = !!args.tls;
+		this.rejectUnauthorized = !!args.rejectUnauthorized;
+		this.password = String(args.password || "");
+		this.username = String(args.username || "");
+		this.realname = String(args.realname || "");
+		this.leaveMessage = String(args.leaveMessage || "");
+		this.sasl = String(args.sasl || "");
+		this.saslAccount = String(args.saslAccount || "");
+		this.saslPassword = String(args.saslPassword || "");
+
+		this.proxyHost = String(args.proxyHost || "");
+		this.proxyPort = parseInt(args.proxyPort, 10);
+		this.proxyUsername = String(args.proxyUsername || "");
+		this.proxyPassword = String(args.proxyPassword || "");
+		this.proxyEnabled = !!args.proxyEnabled;
+
+		// Split commands into an array
+		this.commands = String(args.commands || "")
+			.replace(/\r\n|\r|\n/g, "\n")
+			.split("\n")
+			.filter((command) => command.length > 0);
+
+		// Sync lobby channel name
+		this.channels[0].name = this.name;
+
+		if (this.name !== oldNetworkName) {
+			// Send updated network name to all connected clients
+			client.emit("network:name", {
+				uuid: this.uuid,
+				name: this.name,
+			});
+		}
+
+		if (!this.validate(client)) {
+			return;
+		}
+
+		if (this.irc) {
+			const connected = this.irc.connection && this.irc.connection.connected;
+
+			if (this.nick !== oldNick) {
+				if (connected) {
+					// Send new nick straight away
+					this.irc.changeNick(this.nick);
+				} else {
+					this.irc.user.nick = this.nick;
+
+					// Update UI nick straight away if IRC is not connected
+					client.emit("nick", {
+						network: this.uuid,
+						nick: this.nick,
+					});
+				}
+			}
+
+			if (
+				connected &&
+				this.realname !== oldRealname &&
+				this.irc.network.cap.isEnabled("setname")
+			) {
+				this.irc.raw("SETNAME", this.realname);
+			}
+
+			this.setIrcFrameworkOptions(client);
+
+			if (this.irc.options?.username) {
+				this.irc.user.username = this.irc.options.username;
+			}
+
+			if (this.irc.options?.gecos) {
+				this.irc.user.gecos = this.irc.options.gecos;
+			}
+		}
+
+		client.save();
+	}
+
+	destroy() {
+		this.channels.forEach((channel) => channel.destroy());
+	}
+
+	setNick(this: Network, nick: string) {
+		this.nick = nick;
+		this.highlightRegex = new RegExp(
+			// Do not match characters and numbers (unless IRC color)
+			"(?:^|[^a-z0-9]|\x03[0-9]{1,2})" +
+				// Escape nickname, as it may contain regex stuff
+				_.escapeRegExp(nick) +
+				// Do not match characters and numbers
+				"(?:[^a-z0-9]|$)",
+
+			// Case insensitive search
+			"i"
+		);
+
+		if (this.keepNick === nick) {
+			this.keepNick = null;
+		}
+
+		// TODO: setNick is called in validate() before irc exists. Is that a problem?
```

Comment:
afaik not, because this only updates the option if irc already exists, otherwise it will be created with the new value.